### PR TITLE
Add pre-commit hooks and testing guidelines

### PR DESCRIPTION
**Pull request by codeAesthetic**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,15 @@
...
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "test:coverage": "react-scripts test --coverage",
+    "ci:test": "cross-env CI=true react-scripts test --env=jsdom",
+    "eject": "react-scripts eject",
+    "lint": "eslint src"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "yarn lint && yarn ci:test"
```
Comment (chirgjn) [inline package.json:32]: https://create-react-app.dev/docs/running-tests/#command-line-interface

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -96,3 +98,9 @@ Lowercase is preferred for file and folder names, one exception to this is Files
 - `src/components` will contain reusable UI component for eg. `Button`, `Input` etc.
 - If a style is specific only to the component, and cannot be used anywhere else, add it to `<ComponentName>.styles.ts` in the same folder as the component.
 - All css properties that require number as input should be in multiple of 4 and should be used from `src/common/constants/cssScale.ts`.
+
+### Testing
+
+- Write test cases for each component in their directory for e.g. `src/Home/Home.test.tsx` you can also refer to directory structure.
+- Make sure you cover at least 70% test-case scenario.
+- You can check coverage of your test cases with `yarn test:coverage` command.
```
Comment (chirgjn) [inline CONTRIBUTING.md:106]: We would want to only check coverage for new changes (changes in the current PR), please update the documentation with instructions for the same.

**Comment by chirgjn** [comment]
forked from #9

File: CONTRIBUTING.md (hunk 2 above)
Comment (chirgjn) [reply CONTRIBUTING.md]: *bump*

Clarifying: We want to document the instructions to generate "test coverage" for only the modified files.

**Review by chirgjn — CHANGES_REQUESTED** [review]
PTAL at unresolved comments

**Comment by codeAesthetic** [comment]
@chirgjn I don't see any unresolved comments

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -96,3 +98,10 @@ Lowercase is preferred for file and folder names, one exception to this is Files
 - `src/components` will contain reusable UI component for eg. `Button`, `Input` etc.
 - If a style is specific only to the component, and cannot be used anywhere else, add it to `<ComponentName>.styles.ts` in the same folder as the component.
 - All css properties that require number as input should be in multiple of 4 and should be used from `src/common/constants/cssScale.ts`.
+
+### Testing
+
+- Write test cases for each component in their directory for e.g. `src/Home/Home.test.tsx` you can also refer to directory structure.
+- Make sure you cover at least 70% test-case scenario.
+- You can generate "test coverage" for modified files using `yarn test:coverage` command.
```
Comment (chirgjn) [inline CONTRIBUTING.md:106]: Wouldn't this command generate coverage for the whole project?

If we are running it locally we want to on generate coverage for the new changes.

Comment (codeAesthetic) [reply CONTRIBUTING.md:106]: It generate coverage for new changes

Comment (chirgjn) [reply CONTRIBUTING.md:106]: Can point me to resource for the same. All I was able to find was this: https://create-react-app.dev/docs/running-tests/#coverage-reporting

From this doc it seems like it calculates coverage for the whole project

Comment (codeAesthetic) [reply CONTRIBUTING.md:106]: There is no resource for it but I think it is handled via CRA script. I also couldn't find any relevant resources as such but I ran the command and it check for modified files only.

Comment (chirgjn) [reply CONTRIBUTING.md:106]: Let's remove this from our document for now then, we can add it later after confirming when we have multiple tests.

Comment (codeAesthetic) [reply CONTRIBUTING.md:106]: I confirmed it myself by writing two test files

**Review by chirgjn — APPROVED** [review]
LGTM

**Pull request merged**